src/App.tsx

```diff
@@ -1,222 +1,164 @@
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<string>('all');
+  const [isError, setIsError] = useState('');
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const handleError = (text: string) => {
+    setIsError(text);
+    setTimeout(() => {
+      setIsError('');
+    }, 5000);
+  };
+
+  const handleNewTodos = async () => {
+    if (query.trim().length === 0) {
+      handleError('Type something');
+
+      return;
+    }
+
+    try {
+      const newTodo = await createTodo(user?.id, query);
+
+      setTodos((prevState) => {
+        return [...prevState, newTodo];
+      });
+    } catch (error) {
+      handleError('Can not add a todo');
+    }
+  };
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    try {
+      await deleteTodo(todoId);
+      setTodos(prevState => prevState.filter((a) => a.id !== todoId));
+    } catch (error) {
+      handleError('Can not delete a todo');
+    }
+  }, []);
+
+  const changeTodo = useCallback(async (todoId: number, object: Todo) => {
+    try {
+      const updatedTodo: Todo = await updateTodo(todoId, object);
+
+      setTodos(prevState => (prevState.map((a) => (a.id === todoId
+        ? updatedTodo
+        : a))
+      ));
+    } catch (error) {
+      handleError('Can not update a todo');
+    }
+  }, []);
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    getTodos(user?.id)
+      .then(res => setTodos(res))
+      .catch(() => isError);
+
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case 'active':
+        return todos.filter(todo => !todo.completed);
+
+      case 'completed':
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return [...todos];
+    }
+  }, [todos, filter]);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
         <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
+          {todos.length > 0 && (
+            <button
+              data-cy="ToggleAllButton"
+              type="button"
+              className={classNames(
+                'todoapp__toggle-all',
+                {
+                  'todoapp__toggle-all active':
+                    todos.every(todo => todo.completed),
+                },
+              )}
+            />
+          )}
 
-          <form>
+          <form onSubmit={(event) => {
+            event.preventDefault();
+            handleNewTodos();
+            setQuery('');
+          }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)